### PR TITLE
[LoRA] add an error message when dealing with _best_guess_weight_name ofline

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Potentially fixes: https://github.com/huggingface/diffusers/issues/6089. https://github.com/huggingface/diffusers/issues/6110 is related too. I decided to throw an error rather than trying to engineer a solution. I think that's best here because `_best_guess_weight_name()` solves a particular use case and I don't think we should overengineer it.

### CODE REVIEW

File: src/diffusers/loaders/lora.py [hunk 1]
```diff
@@ -303,7 +311,7 @@ def _best_guess_weight_name(cls, pretrained_model_name_or_path_or_dict, file_ext
             targeted_files = [
                 f for f in os.listdir(pretrained_model_name_or_path_or_dict) if f.endswith(file_extension)
             ]
-        else:
+        elif not local_files_only:
```
Comment (Wauplin) [inline src/diffusers/loaders/lora.py:314]: I don't understand why this change. Given we raise an exception above if `local_files_only`, then it's not needed to check it here, right?

**Review by Wauplin — APPROVED** [review]
Looks good to me @sayakpaul! Thanks for working on this. Agree with you that if another more complete solution already exists then let's not spend too much time maintaining this use case.

File: src/diffusers/loaders/lora.py (hunk 1 above)
Comment (sayakpaul) [reply src/diffusers/loaders/lora.py]: Right. Resolved in 88b93c92.

**Review by patrickvonplaten — APPROVED** [review]
